aiidalab_widgets_base/structures.py

```diff
@@ -33,65 +32,52 @@ class StructureUploadWidget(ipw.VBox):  # pylint: disable=too-many-instance-attr
     frozen = Bool(False)
     DATA_FORMATS = ('StructureData', 'CifData')
 
-    def __init__(  # pylint: disable=too-many-arguments
-            self,
-            text="Upload Structure",
-            storable=True,
-            node_class=None,
-            examples=None,
-            data_importers=None,
-            **kwargs):
+    def __init__(self, importers, storable=True, node_class=None, **kwargs):
         """
-        :param text: Text to display before upload button
-        :type text: str
         :param storable: Whether to provide Store button (together with Store format)
         :type storable: bool
         :param node_class: AiiDA node class for storing the structure.
             Possible values: 'StructureData', 'CifData' or None (let the user decide).
             Note: If your workflows require a specific node class, better fix it here.
         :param examples: list of tuples each containing a name and a path to an example structure
         :type examples: list
-        :param data_importers: list of tuples each containing a name and an object for data importing. Each object
+        :param importers: list of tuples each containing a name and an object for data importing. Each object
         should containt an empty `on_structure_selection()` method that has two parameters: structure_ase, name
-        :type examples: list
-
-        """
-        self.name = None
-        self.file_path = None
-        examples = [] if examples is None else examples
-        data_importers = [] if data_importers is None else data_importers
-        self._structure_sources_tab = ipw.Tab()
-        self.file_upload = FileUploadWidget(text)
-        supported_formats = ipw.HTML(
-            """<a href="https://wiki.fysik.dtu.dk/ase/_modules/ase/io/formats.html" target="_blank">
-            Supported structure formats
-            </a>""")
+        :type examples: list"""
 
-        self.select_example = ipw.Dropdown(options=self.get_example_structures(examples))
-        self.viewer = nglview.NGLWidget()
-        self.btn_store = ipw.Button(description='Store in AiiDA', disabled=True)
-        self.structure_description = ipw.Text(placeholder="Description (optional)")
+        if not isinstance(importers, list):
```

Comment:
It's typically bad practice to expect a certain file type. I assume that any sequence should be fine here? Why insisting on a list? Tuples not ok?

Reply:
Thanks @csadorf, I agree. Also, I was not sure whether to put this check here. I will take it away.